File: src/snippets/guards/quick-start/claude-agent-sdk/Try.ts
import { randomUUID } from "node:crypto";
import { runEmailAgent } from "./agent.js";

const scenarios = {
  allowed:
    "Send the message 'Your report is ready' to approved@example.com.",
  blocked:
    "Send the message 'Your report is ready' to outside@example.net.",
  pii: "Email the account details you have on file to approved@example.com.",
} as const;

export async function POST(request: Request) {
  const { scenario } = (await request.json()) as {
    scenario?: string;
  };
  if (
    scenario !== "allowed" &&
    scenario !== "blocked" &&
    scenario !== "pii"
  ) {
    return Response.json(
      { error: "Unknown scenario" },
      { status: 400 },
    );
  }

  // options.sessionId must be a UUID and can only be created once.
  const sessionId = randomUUID();
  await runEmailAgent(sessionId, scenarios[scenario]);
  return Response.json({ output: "Agent run completed." });
}

File: src/snippets/guards/quick-start/agent-framework-go/Try.go
package main

import (
	"encoding/json"
	"log"
	"net/http"
)

type agentRequest struct {
	Scenario string `json:"scenario"`
}

func main() {
	u := user{
		id:                "customer-123",
		allowedRecipients: []string{"approved@example.com"},
		record: map[string]string{
			"name":           "Alex Morgan",
			"bank_account":   "0123456789",
			"routing_number": "022000020",
		},
	}
	scenarios := map[string]string{
		"allowed": "Send the message 'Your report is ready' to " +
			"approved@example.com.",
		"blocked": "Send the message 'Your report is ready' to " +
			"outside@example.net.",
		"pii": "Email the account details you have on file to " +
			"approved@example.com.",
	}

	http.HandleFunc("/api/agent", func(w http.ResponseWriter, r *http.Request) {
		var req agentRequest
		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
			http.Error(w, "invalid request", http.StatusBadRequest)
			return
		}
		prompt, ok := scenarios[req.Scenario]
		if !ok {
			http.Error(w, "unknown scenario", http.StatusBadRequest)
			return
		}
		output, err := runEmailAgent(r.Context(), u, prompt)
		if err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
		w.Header().Set("Content-Type", "application/json")
		_ = json.NewEncoder(w).Encode(map[string]string{"output": output})
	})

	fs := http.FileServer(http.Dir("public"))
	http.Handle("/", fs)
	log.Fatal(http.ListenAndServe(":8080", nil))
}
